### PR TITLE
Improve language server inlays and symbol highlighting

### PR DESCRIPTION
**Pull request by jabraham17**
Improves the InlayHints for types to work like other symbols. This allows users to hover over the inlay and see documentation information, as well as supporting goto definition/

Also adds symbol highlighting, so if a user selects 1 symbol all of the symbols in the file are also highlighted

[Reviewed by @DanilaFe]

### CODE REVIEW

File: tools/chpl-language-server/src/chpl-language-server.py [hunk 1]
```diff
@@ -100,9 +100,12 @@
     TEXT_DOCUMENT_INLAY_HINT,
     InlayHintParams,
     InlayHint,
+    InlayHintLabelPart,
+    InlayHintKind
 )
 from lsprotocol.types import WORKSPACE_INLAY_HINT_REFRESH
 from lsprotocol.types import TEXT_DOCUMENT_RENAME, RenameParams
+from lsprotocol.types import TEXT_DOCUMENT_DOCUMENT_HIGHLIGHT, DocumentHighlightParams, DocumentHighlight, DocumentHighlightKind
```
Comment (DanilaFe) [inline tools/chpl-language-server/src/chpl-language-server.py:108]: Did you run `black` on this line?

File: tools/chpl-language-server/src/chpl-language-server.py [hunk 2]
```diff
@@ -646,11 +649,12 @@ def _get_param_inlays(
                 position=decl.rng.end,
                 label="param value is " + str(param),
                 padding_left=True,
+                kind=InlayHintKind.Parameter
```
Comment (DanilaFe) [inline tools/chpl-language-server/src/chpl-language-server.py:652]: Dubious -- chapel 'param' is not the same as 'parameter' in conventional usage. 'param annotation' are specifically neither types nor parameters ("parameter annotation" I think is the `f(a=b)` inlay). I deliberately left the `kind` setting out of the inlay data.

**Review by DanilaFe — APPROVED** [review]

File: tools/chpl-language-server/src/chpl-language-server.py (hunk 2 above)
Comment (jabraham17) [reply tools/chpl-language-server/src/chpl-language-server.py:652]: I will pull the InlayHintKind from this PR, it doesn't do much anyways (at least in vscode)

File: tools/chpl-language-server/src/chpl-language-server.py (hunk 1 above)
Comment (jabraham17) [reply tools/chpl-language-server/src/chpl-language-server.py]: I did now 😄

**Pull request merged**